### ISSUE TITLE
chore: change the liveRootPath and archiveRootPath to absolute paths as default

### ISSUE THREAD

**Issue by ata-nas** [hiero-ledger/hiero-block-node#371]
## Story
AS A Block Node Operator/Developer
I WANT my default live/archive roots to be under /opt/hashgraph/blocknode/data
SO THAT I have a meaningful production default

## Tech Notes
- **DEPENDS ON #281**
- the default values for `liveRootPath` and `archiveRootPath` configs of `com.hedera.block.server.persistence.storage.PersistenceStorageConfig` must be set to absolute paths
   - `liveRootPath` -> /opt/hashgraph/blocknode/data/live/
   - `archiveRootPath` -> /opt/hashgraph/blocknode/data/archive/
- see [this comment thread](https://github.com/hashgraph/hedera-block-node/pull/355#discussion_r1861293753)

## What Needs to be Done
- `liveRootPath` production default must be `/opt/hashgraph/data/live`
- `archiveRootPath` production default must be `/opt/hashgraph/blocknode/data/archive`
- migrate from `String` type configs to `Path` type configs
- simplify/remove preconditions and path resolution logic in constructor
- configuration does not create any roots, it is simply value holder and has no logic

**Comment by jsync-swirlds** [comment]
Perhaps, rather than setting absolute paths for each entry, create a configuration for "blockNode.filesystem.basePath" set to `/opt/hashgraph/blocknode/data` and then the two storage paths can be relative to that.

**Comment by mattp-swirldslabs** [comment]
With the deterministic build changes, we'll likely need to adjust the Dockerfile directives to create any new directories (opt) and to ensure the permissions are switched over. Until we get an GHA trigger to dry-run push images, the best way to make sure your branch is still deterministic is to run:  `gh workflow run .github/workflows/release-push-image.yaml --ref <your branch name>` or manually trigger the "Release Workflow" GHA after pointing the Run Workflow dropdown to your branch
![image](https://github.com/user-attachments/assets/6ab5fb1c-e298-47b0-a1d0-a0bc43fce071)

**Comment by ata-nas** [comment]
> With the deterministic build changes, we'll likely need to adjust the Dockerfile directives to create any new directories (opt) and to ensure the permissions are switched over. Until we get an GHA trigger to dry-run push images, the best way to make sure your branch is still deterministic is to run: `gh workflow run .github/workflows/release-push-image.yaml --ref <your branch name>` or manually trigger the "Release Workflow" GHA after pointing the Run Workflow dropdown to your branch ![image](https://github.com/user-attachments/assets/6ab5fb1c-e298-47b0-a1d0-a0bc43fce071)

@mattp-swirldslabs that would be a good item to do, by default, with the docker image built as is, the user that runs the jvm has no perms for `/opt`. Do you have some open/pending item that could include the creation of `/opt` if needed and to ensure the user that would run the jvm inside the docker would have the sufficient perms? Would help a lot :)

**Comment by ata-nas** [comment]
> Perhaps, rather than setting absolute paths for each entry, create a configuration for "blockNode.filesystem.basePath" set to `/opt/hashgraph/blocknode/data` and then the two storage paths can be relative to that.

I have made a separate issue for this base path. Reason is if we want to have the base root to be resolved as `blockNode.filesystem.basePath` we need a separate config record for it. It also makes sense that we have a base block node config, it is not only the persistence that will need this base path potentially (and other configs which may reside in the base config record). This will require some additional injection and will complicate the PR (and fix/migrate some tests as well), so to keep it simple, for the PR that will resolve this issue, will be introduced both roots to be under `/opt` by default for now.
